### PR TITLE
ORM 2.5 and upwards to 2.11

### LINKED ISSUE #335: Move to newer version of Doctrine (2.5 at least)
The version of Doctrine we currently use is no longer supported (https://www.doctrine-project.org/projects/orm.html). We were stuck on it because the next version (2.5) required php 5.4  or greater. With #238 we removed php 5.3 support and so can now migrate upwards. This may be true for other components as well.

Pretty careful testing would be needed

### PR DESCRIPTION
**Pull request by ineilson**
Closes #335

### CODE REVIEW

File: tests/doctrine/DoctrineCleanInsert1Test.php [hunk 1]
```diff
@@ -182,11 +182,11 @@ public function testShowDoctrineReadFailiureUntilCommitted(){
     $this->em->persist($site);
     // After persist, Doctrine querying does not return our site until we flush, hence we get 0
     // returned from our repository.
-    $this->assertEquals(0, count($this->em->getRepository('Site')->findOneBy(array('shortName' => 'site' . $n))));
+    $this->assertEquals(0, count($this->em->getRepository('Site')->findBy(array('shortName' => 'site' . $n))));
```
Comment (gregcorbett) [inline tests/doctrine/DoctrineCleanInsert1Test.php:185]: What was the error here? It [looks like](https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/tutorials/getting-started.html) the findOneBy methond still exists in Doctrine 2.11...

Comment (ineilson) [reply tests/doctrine/DoctrineCleanInsert1Test.php:185]: image
Something got tightened up.

File: composer.json [hunk 2]
```diff
@@ -2,12 +2,13 @@
     "name": "stfc-egi/gocdb",
     "description": "A web portal and REST style API for e-Infrastructure topology managment",
     "require": {
-	"doctrine/orm": "2.4.8",
-	"doctrine/dbal": "2.5.4"
-    }, 
+	"doctrine/orm": "2.11.*",
+    "symfony/cache": "*",
+    "doctrine/annotations": "*"
```
[the remark this answers is not in the document]
Comment (ineilson) [reply composer.json:7]: For the cache - https://github.com/doctrine/orm/blob/2.10.x/UPGRADE.md#minor-bc-break-setup-tool-needs-cache-implementation

[the remark this answers is not in the document]
Comment (ineilson) [reply composer.json:7]: For the annotations - I'm less clear exactly what and why it happens, but I'm not alone and the explanation here suggests the 'fix' (and a possible change, which I haven't implemented.) https://github.com/doctrine/orm/issues/9449

**Comment by ineilson** [comment]
> Looking at the unit tests, I presume we can only use Doctrine 2.6+ if we are using PHP 7.4?

More accurately PHP 7.2+ I think.

File: composer.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (gregcorbett) [reply composer.json:7]: Can you capture both these in ammended commit messages?

**Comment by gregcorbett** [comment]
> > Looking at the unit tests, I presume we can only use Doctrine 2.6+ if we are using PHP 7.4?
> 
> More accurately PHP 7.2+ I think.

👍 , I assume as we climb to the Doctrine 2.11 ladder that get's tightened up to PHP 7.4+ - and IMO we might as well go straight to PHP 7.4 rather than stop off at PHP 7.2

**Comment by gregcorbett** [comment]
Could you open a seperate PR with just the ORM 2.5 and PHP 7.4 changes (so the first four commits)? I think I'll want to split ORM 2.5, PHP 7.4 and ORM 2.11 across seperate releases.

File: composer.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ineilson) [reply composer.json:7]: Ammended - https://github.com/GOCDB/gocdb/pull/344/commits/cb4ac105004723e999c4392b9c97eca283096e92

**Comment by ineilson** [comment]
> I think I'll want to split ORM 2.5, PHP 7.4 and ORM 2.11 across seperate releases.

Is there a particular reason to do this in two pieces?

File: composer.json [hunk 3]
```diff
@@ -2,12 +2,13 @@
     "name": "stfc-egi/gocdb",
     "description": "A web portal and REST style API for e-Infrastructure topology managment",
     "require": {
-	"doctrine/orm": "2.4.8",
-	"doctrine/dbal": "2.5.4"
-    }, 
+	"doctrine/orm": "2.11.*",
```
Comment (gregcorbett) [inline composer.json:5]: This wild card is probably fine, as the doctrine ORM will be held to a specific version by composer.lock unless that file is regenrated manually.

Rolling this out will require some thought from @gregcorbett.

**Review by gregcorbett — APPROVED** [review]

**Pull request closed without merging**